### PR TITLE
added qt5 support with poppler

### PR DESCRIPTION
**Pull request by DrunkenMonkey85**
added qt5 support to the poppler lib

### CODE REVIEW

File: Library/Formula/poppler.rb [hunk 1]
```diff
@@ -51,8 +53,12 @@ def install
 
     if build.with? "qt"
       args << "--enable-poppler-qt4"
+    #edded to support qt5
```
Comment (bfontaine) [inline Library/Formula/poppler.rb]: You can remove this line.

File: Library/Homebrew/test/Gemfile.lock [hunk 2]
```diff
@@ -14,3 +14,6 @@ DEPENDENCIES
   minitest (~> 5.3)
   mocha (~> 1.1)
   rake (~> 10.3)
+
+BUNDLED WITH
+   1.10.2
```
Comment (bfontaine) [inline Library/Homebrew/test/Gemfile.lock]: You can remove these lines.

File: Library/Formula/poppler.rb [hunk 3]
```diff
@@ -51,8 +53,12 @@ def install
...
+    elsif build.with? "qt5"
+      args << "--enable-poppler-qt5"
+      ENV.append_path "PKG_CONFIG_PATH" , "#{HOMEBREW_PREFIX}/opt/qt5/lib/pkgconfig"
```
Comment (DomT4) [inline Library/Formula/poppler.rb]: Let's use `#{Formula["qt5"].opt_lib}/pkgconfig` here instead.

File: Library/Formula/poppler.rb [hunk 4]
```diff
@@ -12,6 +12,7 @@ class Poppler < Formula
   end
 
   option "with-qt", "Build Qt backend"
+  option "with-qt5", "Build Qt5 backend" #added to support qt5
```
Comment (mikemcquaid) [inline Library/Formula/poppler.rb]: Remove the comment, thanks.

File: Library/Formula/poppler.rb [hunk 5]
```diff
@@ -30,6 +31,7 @@ class Poppler < Formula
   depends_on 'openjpeg'
 
   depends_on "qt" => :optional
+  depends_on "qt5" => :optional         #added to support qt5
```
Comment (mikemcquaid) [inline Library/Formula/poppler.rb]: Remove the comment, thanks.

File: Library/Formula/poppler.rb [hunk 6]
```diff
@@ -51,8 +53,12 @@ def install
 
     if build.with? "qt"
       args << "--enable-poppler-qt4"
+    #edded to support qt5
+    elsif build.with? "qt5"
+      args << "--enable-poppler-qt5"
+      ENV.append_path "PKG_CONFIG_PATH" , "#{HOMEBREW_PREFIX}/opt/qt5/lib/pkgconfig"
     else
-      args << "--disable-poppler-qt4"
+      args << "--disable-poppler-qt4 --disable-poppler-qt5"
```
Comment (mikemcquaid) [inline Library/Formula/poppler.rb]: `args << "--disable-poppler-qt4" << "--disable-poppler-qt5"`

File: Library/Formula/poppler.rb [hunk 7]
```diff
@@ -62,3 +68,4 @@ def install
     resource('font-data').stage { system "make", "install", "prefix=#{prefix}" }
   end
 end
+
```
Comment (mikemcquaid) [inline Library/Formula/poppler.rb]: Remove this line.

Comment (DrunkenMonkey85) [inline Library/Formula/poppler.rb]: Okay... but the brew test tools proposed to add an empty line

File: Library/Formula/poppler.rb (hunk 1 above)
Comment (DrunkenMonkey85) [inline Library/Formula/poppler.rb]: Done

File: Library/Homebrew/test/Gemfile.lock (hunk 2 above)
Comment (DrunkenMonkey85) [inline Library/Homebrew/test/Gemfile.lock]: Done

File: Library/Formula/poppler.rb (hunk 3 above)
Comment (DrunkenMonkey85) [inline Library/Formula/poppler.rb]: Done... That looks pretty nice imo

File: Library/Formula/poppler.rb (hunk 4 above)
Comment (DrunkenMonkey85) [inline Library/Formula/poppler.rb]: done

File: Library/Formula/poppler.rb (hunk 5 above)
Comment (DrunkenMonkey85) [inline Library/Formula/poppler.rb]: done

File: Library/Formula/poppler.rb (hunk 6 above)
Comment (DrunkenMonkey85) [inline Library/Formula/poppler.rb]: done :)

**Comment by DrunkenMonkey85** [comment]
i recently pushed the changes to my upstream... is there something more i need to do?

File: Library/Formula/poppler.rb [hunk 8]
```diff
@@ -61,4 +66,4 @@ def install
     system "make install"
     resource('font-data').stage { system "make", "install", "prefix=#{prefix}" }
   end
-end
+end
```
Comment (DomT4) [inline Library/Formula/poppler.rb]: Need a new, empty line at the very end of the formula.

**Comment by DrunkenMonkey85** [comment]
Hey guys! 
do i still have anything to do?? the jenkins build seems to fail but i'm not sure what if this failure is related my changes.

**Comment by bfontaine** [comment]
`mat` depends on `poppler` and is failing here:

```
Testing mat
==> /usr/local/Cellar/mat/0.5.2/bin/mat -l
/usr/local/Cellar/mat/0.5.2/bin/mat: /usr/local/Cellar/mat/0.5.2/libexec/bin/mat: /usr/local/opt/python/bin/python2.7: bad interpreter: No such file or directory
```

There’s a PR to update `mat` (#38600) but it also have issues.

**Comment by DomT4** [comment]
We should ignore `mat` failures for now. I'm actually kind of tempted to boneyard it until we get a response upstream to some outstanding issues.

**Comment by bfontaine** [comment]
We had a response [here](https://mailman.boum.org/pipermail/mat-dev/2015-April/000104.html), I’d like to try a patch before we boneyard it.

**Comment by DomT4** [comment]
Sure, feel free.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by DrunkenMonkey85** [comment]
Thank you for creating such an awesome project! it was a pleasure to contribute :-) :+1:

**Pull request closed without merging**